server/src/test/java/org/elasticsearch/common/compress/DeflateCompressTests.java

```diff
@@ -361,6 +364,43 @@ public void run() {
         }
     }
 
+    public void testCompressUncompressWithCorruptions() throws Exception {
+        final Random r = random();
+        for (int i = 0; i < 10; i++) {
+            byte[] bytes = new byte[TestUtil.nextInt(r, 1, 100000)];
+            r.nextBytes(bytes);
+            final var offset = between(0, bytes.length - 1);
+            final var length = between(0, bytes.length - offset);
+            final var original = new BytesArray(bytes, offset, length);
+            final var compressed = compressor.compress(original);
+
+            if (randomBoolean()) {
+                var corruptIndex = between(0, compressed.length() - 1);
+                BytesRef bytesRef;
+                final var iterator = compressed.iterator();
+                while ((bytesRef = iterator.next()) != null) {
+                    if (corruptIndex < bytesRef.length) {
+                        bytesRef.bytes[bytesRef.offset + corruptIndex] = randomValueOtherThan(
+                            bytesRef.bytes[bytesRef.offset + corruptIndex],
+                            () -> (byte) (r.nextInt() & 0xff)
+                        );
+                        break;
+                    } else {
+                        corruptIndex -= bytesRef.length;
+                    }
+                }
+                try {
+                    compressor.uncompress(compressed);
+                } catch (ZipException e) {
+                    // ok
+                }
```

Comment:
Should this fail if exception was not thrown?

Reply:
No, there's no checksum or anything so modifying a byte will sometimes yield a valid (but different) output.